libparsec/crates/client/src/client/mod.rs

```diff
@@ -533,6 +539,165 @@ impl Client {
+        &self,
+        new_device: LocalDevice,
+        access: DeviceAccessStrategy,
+    ) -> Result<AvailableDevice, ImportRecoveryDeviceError> {
+        // save recovery device
+        let saved_device = save_device(&self.config.config_dir, &access, &new_device).await?;
+
+        let outcome = self
+            .certificates_ops
+            .create_new_device(new_device, self.device.clone())
+            .await?;
+
+        let latest_known_timestamps = match outcome {
+            CertificateBasedActionOutcome::LocalIdempotent => return Ok(saved_device), // not sure how it could happen though
+            CertificateBasedActionOutcome::Uploaded {
+                certificate_timestamp,
+            }
+            | CertificateBasedActionOutcome::RemoteIdempotent {
+                certificate_timestamp,
+            } => PerTopicLastTimestamps::new_for_common_and_shamir(
+                certificate_timestamp,
+                certificate_timestamp,
+            ),
+        };
+        self.certificates_ops
+            .poll_server_for_new_certificates(Some(&latest_known_timestamps))
+            .await
+            .map_err(|e| match e {
+                CertifPollServerError::Stopped => ImportRecoveryDeviceError::Stopped,
+                CertifPollServerError::Offline => ImportRecoveryDeviceError::Offline,
+                CertifPollServerError::InvalidCertificate(err) => {
+                    ImportRecoveryDeviceError::InvalidCertificate(err)
+                }
+                CertifPollServerError::Internal(err) => err
+                    .context("Cannot poll server for new certificates")
+                    .into(),
+            })?;
+
+        Ok(saved_device)
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+
+pub enum ExportRecoveryDeviceError {
+    #[error("Component has stopped")]
+    Stopped,
+    #[error("Cannot reach the server")]
+    Offline,
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+    #[error("User {0} is revoked")]
+    UserRevoked(UserID),
+
+    #[error("Our clock ({client_timestamp}) and the server's one ({server_timestamp}) are too far apart")]
+    TimestampOutOfBallpark {
+        server_timestamp: DateTime,
+        client_timestamp: DateTime,
+        ballpark_client_early_offset: f64,
+        ballpark_client_late_offset: f64,
+    },
+    #[error(transparent)]
+    InvalidCertificate(#[from] Box<InvalidCertificateError>),
+    #[error(transparent)]
+    DataError(#[from] libparsec_types::DataError),
+    // #[error(transparent)]
+    // EncryptionError(#[from] CertifEncryptForUserError),
+    #[error(transparent)]
+    GetCertificateError(#[from] GetCertificateError),
+    #[error(transparent)]
+    CertifDeviceError(#[from] CertifDeviceError),
+    #[error(transparent)]
+    PlatformExportRecoveryDeviceError(#[from] PlatformExportRecoveryDeviceError),
+    #[error(transparent)]
+    ConnectionError(#[from] ConnectionError),
+}
+
+#[derive(Debug, thiserror::Error)]
+
```

Comment:
```suggestion
#[derive(Debug, thiserror::Error)]
```

Remove extra newline